chapters/annotations.tex

```diff
@@ -336,15 +336,46 @@ \section{Annotations for Code Generation}\label{annotations-for-code-generation}
 "Evaluate" "=" ( false | true )
 \end{lstlisting}\end{synopsis}
 \begin{semantics}
-The annotation \lstinline!Evaluate! can occur in the component declaration, its type declaration, or a base-class of the type-declaration.  In the case of multiple conflicting annotations it is handled similarly to modifiers (e.g., an \lstinline!Evaluate! annotation on the component declaration takes precedence).  The annotation \lstinline!Evaluate! only has effect for a component declared with the prefix \lstinline!parameter!.
+The annotation \lstinline!Evaluate! can occur in the component declaration, its type declaration, or a base-class of the type-declaration.
+In the case of multiple conflicting annotations it is handled similarly to modifiers (e.g., an \lstinline!Evaluate! annotation on the component declaration takes precedence).
+The annotation \lstinline!Evaluate! is only allowed for parameters and constants.
 
-If \lstinline!Evaluate = true!, the model developer proposes to utilize the value for the symbolic processing. In that case, it is not possible to change the parameter value after symbolic pre-processing.
+Setting \lstinline!Evaluate = true! for a structural parameter, means that its value must be determined during translation, similar to a constant.
+For a normal parameter, it has no impact, and it is recommended to issue a warning (except when the parameter is normal due to dependency on a parameter with \lstinline!Evaluate = false!, as this could be a sign of intentional overriding of \lstinline!Evaluate = true!, see example below).
```

Comment:
The proposal tries to give more imperative meaning to `Evaluate`, and with the separation of structural and normal parameters it has now become possible to be more precise in the different cases.  The intention here is that in the case of "except when…", you disregard the "and…" part of the sentence; `Evaluate = true` simply has no meaning in this case.  Because it is generally counter-intuitive that it has no meaning, a warning is recommended.

(The old way of defining `Evaluate` in terms of just "proposing" things was vague enough to avoid even getting attention to this sort of questions.)

Reply:
> I think at the very least it should be said somewhere that if there are cycles like in the example with p and q above such bindings must be considered not as bindings but as equations and added by the tool to the initial equation section. However, I am strongly against this idea in general.

I'll try to respond to this below, when explaining concerns regarding the poll alternatives.

Reply:
This conversation is very old and outdated in view of the last design meeting discussions.  Resolving.